jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketChannel.java

```diff
@@ -255,6 +262,7 @@ public boolean isClosed()
     public void setWebSocketConnection(WebSocketConnection connection)
     {
         this.connection = connection;
+        getConnection().getEndPoint().setIdleTimeout(idleTimeout.toMillis());
```

Comment:
I would prefer if `idleTimeout` was initialised to `null` and the set done here only if it is non null.... and then null the field after the set.